build.sbt

```diff
@@ -21,44 +22,121 @@ import sbt._
+// these are retrieved during CodeBuild steps
+val awsSharedRepoPass = sys.props.get("ci.internalCentralMvnPassword").getOrElse("")
+val internalReleaseRepoPass = sys.props.get("ci.internalTeamMvnPassword").getOrElse("")
+// remove the PATCH portion of the spark version number for use in release binary
+// e.g. MAJOR.MINOR.PATCH => MAJOR.MINOR
+val releaseSparkVersion = testSparkVersion.substring(0, testSparkVersion.lastIndexOf("."))
+
+def incompatibleSparkVersions(): FileFilter = {
+  val versionArray = testSparkVersion.split("""\.""").map(Integer.parseInt)
+  val major = versionArray(0)
+  val minor = versionArray(1)
+  ("*_spark_*_*_*" -- s"*_spark_${major}_${minor}_*")
+}
+
+def ciPipelineSettings[P](condition: Boolean): Seq[Def.Setting[_]] = {
+  if (condition) {
+    val (fetchRealm, publishRealm, fetchUrl, publishUrl, fetchRepo, publishRepo, userName) =
+      try {
+        val prop = new Properties()
+        prop.load(new FileInputStream("ci/internal_ci.properties"))
+        (
+          prop.getProperty("fetchRealm"),
+          prop.getProperty("publishRealm"),
+          prop.getProperty("fetchUrl"),
+          prop.getProperty("publishUrl"),
+          prop.getProperty("fetchRepo"),
+          prop.getProperty("publishRepo"),
+          prop.getProperty("userName")
+        )
+      } catch {
+        case e: Exception =>
+          e.printStackTrace()
+          sys.exit(1)
+      }
+    Seq(
+      credentials := Seq(
+        Credentials(fetchRealm, fetchUrl, userName, awsSharedRepoPass),
+        Credentials(publishRealm, publishUrl, userName, internalReleaseRepoPass)
+      ),
+      resolvers := Seq(fetchRealm at fetchRepo),
+      externalResolvers := Seq(fetchRealm at fetchRepo),
+      publishTo := Some (publishRealm at publishRepo),
+      pomIncludeRepository := { (_: MavenRepository) => false },
+      releaseProcess := Seq[ReleaseStep](publishArtifacts)
+    )
+  }
+  else Seq(
+    publishTo := {
+      val nexus = "https://oss.sonatype.org/"
+      if (isSnapshot.value)
+        Some("snapshots" at nexus + "content/repositories/snapshots")
+      else
+        Some("releases"  at nexus + "service/local/staging/deploy/maven2")
+    },
+    // Add publishing to spark packages as another step.
+    releaseProcess := Seq[ReleaseStep](
+      checkSnapshotDependencies,
+      inquireVersions,
+      runTest,
+      setReleaseVersion,
+      commitReleaseVersion,
+      tagRelease,
+      publishArtifacts,
+      setNextVersion,
+      commitNextVersion,
+      pushChanges
+    )
+  )
+}
 
 lazy val root = Project("spark-redshift", file("."))
+  .enablePlugins(BuildInfoPlugin)
   .configs(IntegrationTest)
   .settings(net.virtualvoid.sbt.graph.Plugin.graphSettings: _*)
   .settings(Project.inConfig(IntegrationTest)(rawScalastyleSettings()): _*)
   .settings(Defaults.coreDefaultSettings: _*)
   .settings(Defaults.itSettings: _*)
+  .settings(ciPipelineSettings(isCI))
```

Comment:
What is isCI? Is this for internal Amazon release pipeline settings?

Reply:
Yes, it is a code path for our internal Continuous Integration build (CI). It is disabled by default.